sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/SchemaRegistryAvroSerializerBuilder.java

```diff
@@ -0,0 +1,112 @@
+     * If not auto-registering, serializer will request schema ID for matching data schema under specified group.
+     *
+     * @param schemaGroup Azure Schema Registry schema group
+     * @return updated {@link SchemaRegistryAvroSerializerBuilder} instance
+     */
+    public SchemaRegistryAvroSerializerBuilder schemaGroup(String schemaGroup) {
+        this.schemaGroup = schemaGroup;
+        return this;
+    }
+
+    /**
+     * Specifies authentication behavior with Azure Schema Registry
+     * @param credential TokenCredential to be used to authenticate with Azure Schema Registry service
+     * @return updated {@link SchemaRegistryAvroSerializerBuilder} instance
+     */
+    public SchemaRegistryAvroSerializerBuilder credential(TokenCredential credential) {
+        this.credential = credential;
+        return this;
+    }
+
+    /**
+     * If specified true, serializer will register schemas against Azure Schema Registry service under the specified
+     * group.  See Azure Schema Registry documentation for a description of schema registration behavior.
+     *
+     * If specified false, serializer will simply query the service for an existing ID given schema content.
+     * Serialization will fail if the schema has not been pre-created.
+     *
+     * Auto-registration is **NOT RECOMMENDED** for production scenarios.
+     *
+     * @param autoRegisterSchemas flag for schema auto-registration
+     * @return updated {@link SchemaRegistryAvroSerializerBuilder} instance
+     */
+    public SchemaRegistryAvroSerializerBuilder autoRegisterSchema(boolean autoRegisterSchemas) {
+        this.autoRegisterSchemas = autoRegisterSchemas;
+        return this;
+    }
+
+    /**
+     * Specifies maximum schema object cache size for underlying CachedSchemaRegistryClient.  If specified cache
+     * size is exceeded, all caches are recycled.
+     *
+     * @param maxSchemaMapSize maximum number of schemas per cache
+     * @return updated {@link SchemaRegistryAvroSerializerBuilder} instance
+     */
+    public SchemaRegistryAvroSerializerBuilder maxSchemaMapSize(int maxSchemaMapSize) {
+        this.maxSchemaMapSize = maxSchemaMapSize;
+        return this;
+    }
+
+    /**
+     * Instantiates SchemaRegistry
+     * @return {@link SchemaRegistryAvroAsyncSerializer} instance
+     *
+     * @throws NullPointerException if parameters are incorrectly set.
+     * @throws IllegalArgumentException if credential is not set.
+     */
+    public SchemaRegistryAvroAsyncSerializer buildAsyncClient() {
+        return new SchemaRegistryAvroAsyncSerializer(this.buildSyncClient());
+    }
+
+    /**
+     * Instantiates {@link SchemaRegistryAvroSerializer}
+     * @return {@link SchemaRegistryAvroSerializer} instance
+     *
+     * @throws NullPointerException if parameters are incorrectly set.
+     * @throws IllegalArgumentException if credential is not set.
+     */
+    public SchemaRegistryAvroSerializer buildSyncClient() {
```

Comment:
Validate inputs to check if all required fields like registryUrl and credential are set.

Reply:
Moved CachedSchemaRegistryClient build call to this builder so it can take advantage of the validation already in the client builder.